### PR TITLE
Added enhanced Maximum number of Subflows Considerations

### LINKED ISSUE #306: MP_MAXNUM_SUBFLOW
I understand we can create infinite number of subflows in a MP-DCCP connection and there is a recommendation limit the number of subflows. However, would it not be better to define a MP feature MP_MAXNUM_SUBFLOW to exchange the imposed limit to the peers,so that it is much clearner to put restrictions via signaling on the connection and save the client from trying to creat new subflows and server from rejecting it?

If this dicusison has already happend please point me the conclusion(s).

### PR DESCRIPTION
**Pull request by markusa**
Closes #306

### CODE REVIEW

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 1]
```diff
@@ -1543,11 +1543,13 @@ for each path allows.
...
 
-## Maximum number of Subflows
+## Maximum number of Subflows Considerations
 
 In theory, an infinite number of subflows can be created within an MP-DCCP connection, as there is no element in the protocol that represents a restriction. In practical scenarios, however, there will be resource limitations on the host or use cases that do not benefit from additional subflows.
 
-It is RECOMMENDED to limit the number of subflows in implementations and to reject incoming subflow requests with a DCCP-Reset using the Reset Code "too busy" as specified in {{RFC4340}} if this limit is exceeded.
+In typical multi-homing scenarios such as use in a smartphone or a hybrid access gateway (host1), the access paths are limited to two connections cellular and WiFi/fixed. Depending on the path usage strategy ({{path_usage_strategy}}), this leads to the creation of no more than 2-4 subflows, depending on whether the peer (host2) is equipped with one or two network interfaces and the maximum path resources are to be utilized (N interfaces host1 x M interfaces host2). Anything beyond this is unlikely to add value and the creation of further subflows SHOULD be avoided.
```
Comment (zaheduzzaman) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1550]: This paragraph seems not required.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 2]
```diff
@@ -1543,11 +1543,13 @@ for each path allows.
...
+
+It is RECOMMENDED to limit the number of subflows in implementations and to reject incoming subflow requests with a DCCP-Reset using the Reset Code “too busy” according to {{RFC4340}} if the resource limit is exceeded or it is known that the multipath connection will not benefit from further subflows. Likewise, the host that wants to create the subflows is RECOMMENDED to consider the aspect of available resources and the possible gains.
```
Comment (zaheduzzaman) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1552]: I think not only the number of subflow needs consideration but also rate of subflow creation needs considerations. MPTCP has heuristics for delayed subflow start, inspiration can be taken from there.

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 1 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: Done in https://github.com/tsvwg/ietf-multipath-dccp/pull/315/commits/c9f48858cfe0f83a2dd632656e1bbaca4a3b7739

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 2 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1550]: Done in https://github.com/tsvwg/ietf-multipath-dccp/pull/315/commits/702ee705189b48ca7f9143a7cc30b3065b288016

**Pull request merged**